src/textual/css/model.py

```diff
@@ -94,6 +94,13 @@ def _check_universal(self, node: DOMNode) -> bool:
     def _check_type(self, node: DOMNode) -> bool:
         if self.name not in node._css_type_names:
             return False
+
+        if "last-child" in self.pseudo_classes and node.parent.children[-1] == node:
```

Comment:
this and the first-child can fail if the `node` has no `parent`. Is this possible in textual?
If yes, what would be the expected behavior in such case? Should the root node matches the `last-child` or `first-child` pseudo elements?

Reply:
All widgets in the DOM always have a parent.
So, this is fine as long as this only gets called after the widget has been mounted.

Reply:
The `app` is at the root of the DOM, and it doesn't have a parent.

Reply:
@willmcgugan I always forget that the app is a widget 🙃
In that case, what's the reply you want to give to csernazs?